src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs

```diff
@@ -0,0 +1,379 @@
+                    }
+                    Log.LogError("Property {0} not found in {1}", property.Name, jsonFilePath);
+                    throw new Exception();
+                }
+                else
+                {
+                    if (jsonModel?.Items != null && jsonModel.Items.TryGetValue(property.Name, out var itemModels))
+                    {
+                        return ConvertItems (itemModels);
+                    }
+
+                }
+                return null;
+            }
+
+            public static ITaskItem[] ConvertItems(JsonModelItem[] itemModels)
+            {
+                var items = new ITaskItem[itemModels.Length];
+                for (int i = 0; i < itemModels.Length; i++)
+                {
+                    var itemModel = itemModels[i];
+                    var item = new TaskItem(itemModel.Identity);
+                    if (itemModel.Metadata != null)
+                    {
+                        foreach (var metadata in itemModel.Metadata)
+                        {
+                            if (string.Equals(metadata.Key, "Identity", StringComparison.OrdinalIgnoreCase))
+                                continue;
+                            item.SetMetadata(metadata.Key, metadata.Value);
+                        }
+                    }
+                    items[i] = item;
+                }
+                return items;
+            }
+
+            public void SetPropertyValue (TaskPropertyInfo property, object? value)
+            {
+                if (logDebugTask) Log.LogMessage("SetPropertyValue called with {0}", property.Name);
+                if (property.Name == "JsonFilePath")
+                {
+                    jsonFilePath = (string)value!;
+                }
+                else
+                    throw new Exception ($"JsonToItemsTask {TaskName} cannot set property {property.Name}");
+            }
+
+        }
+
+        public class JsonModelRoot
+        {
+            [JsonConverter(typeof(CaseInsensitiveDictionaryConverter))]
+            public Dictionary<string, string>?  Properties {get; set;}
+            public Dictionary<string, JsonModelItem[]>? Items {get; set;}
+
+            public JsonModelRoot() {}
+        }
+
+        [JsonConverter(typeof(JsonModelItemConverter))]
+        public class JsonModelItem
+        {
+            public string? Identity {get; set;}
+            // n.b. will  be deserialized case insensitive
+            public Dictionary<string, string>? Metadata {get;  set; }
+        }
+
+        public class CaseInsensitiveDictionaryConverter : JsonConverter<Dictionary<string, string>>
+        {
+            public override Dictionary<string, string> Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                var dict = JsonSerializer.Deserialize<Dictionary<string, string>>(ref reader, options);
+                if (dict == null)
+                    return null!;
```

Comment:
Should the uses of JsonSerializer.Deserialize* also catch `JsonException`?

Reply:
I don't think so - we want it to propagate

Reply:
Ok, then we should catch this up the chain somewhere, so we can log it as an error with the filename.